src/annotator/config/index.js

```diff
@@ -1,53 +1,240 @@
+      },
+      notebookAppUrl: {
+        allowInBrowserExt: true,
+        defaultValue: null,
+        valueFn: () => this.settings.notebookAppUrl,
+      },
+      sidebarAppUrl: {
+        allowInBrowserExt: true,
+        defaultValue: null,
+        valueFn: () => this.settings.sidebarAppUrl,
+      },
+      // Sub-frame identifier given when a frame is being embedded into
+      // by a top level client
+      subFrameIdentifier: {
+        allowInBrowserExt: true,
+        defaultValue: null,
+        valueFn: this.settings.hostPageSetting,
+      },
+      externalContainerSelector: {
+        allowInBrowserExt: false,
+        defaultValue: null,
+        valueFn: this.settings.hostPageSetting,
+      },
+    };
+  }
+
+  annotatorDefinitions() {
+    return this.configDefinitions();
+  }
+
+  sidebarDefinitions() {
+    return this.configDefinitions();
+  }
+
+  notebookDefinitions() {
+    const config = this.configDefinitions();
+    // Don't pass `annotations` setting to notebook.
+    delete config.annotations;
+    return config;
+  }
+
+  /**
+   * Process the config definition to create the final config value.
+   *   - find the value from its source (valueFn)
+   *   - filter allowInBrowserExt as needed
+   *   - coerce the value if necessary
+   *   - set a default if the value is undefined
+   * @param {string} name - Config key name
+   * @param {ConfigDefinition} configDef
+   */
+  createValueFromDefinition(name, configDef) {
+    // If allowInBrowserExt is false and this is the browser extension context, skip value
+    if (!configDef.allowInBrowserExt && this.isBrowserExtension) {
+      return undefined;
+    }
+
+    const coerceFn = configDef.coerce ? configDef.coerce : name => name; // use no-op if omitted
+    // Get the value from the configuration source and run through an optional coerce method
+    const value = coerceFn(configDef.valueFn.call(this, name));
+
+    // If a defaultValue is provided and the value is undefined, set the default
+    if (value === undefined && configDef.defaultValue !== undefined) {
+      return configDef.defaultValue;
+    } else {
+      return value;
+    }
+  }
+
+  /**
+   * Return the configuration for a given application context
+   *
+   * @param {'sidebar'|'notebook'|'annotator'} [appContext] -
+   *   The name of the app. Omit this param to return the total configuration
+   */
+  getConfig(appContext = 'annotator') {
+    const config = {};
+
+    // Filter the config based on the application context as some config values
+    // may be inappropriate or erroneous for some applications.
+    let partialConfigDef = {};
+    switch (appContext) {
+      case 'annotator':
+        partialConfigDef = this.annotatorDefinitions();
+        break;
+      case 'sidebar':
+        partialConfigDef = this.sidebarDefinitions();
+        break;
+      case 'notebook':
+        partialConfigDef = this.notebookDefinitions();
+        break;
```

Comment:
There's no `default` case here, which means that an unrecognized `appContext` would fall through (static type checking would catch this based on the typedef, but not at runtime). Suggest a `default` case that throws an Error (unrecognized configuration context).